src/blk/kernel/KernelDevice.h

```diff
@@ -150,7 +150,7 @@ class KernelDevice : public BlockDevice,
 		bool buffered,
 		int write_hint = WRITE_LIFE_NOT_SET) override;
   int flush() override;
-  int _discard(uint64_t offset, uint64_t len);
+  int _discard(uint64_t offset, uint64_t len) override;
```

Comment:
Current implementation of this func doesn't check if device supports discards. Please add this check.

Reply:
done